### PR TITLE
Reorganize test/unit/api directory.

### PR DESCRIPTION
**Pull request by aniemetz**
This moves .cpp files to directory test/unit/api/cpp and python files in
test/python/unit/api to test/unit/api/python.

@yoni206 this will affect #7597.

### CODE REVIEW

File: test/unit/api/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+# Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+# in the top-level source directory and their institutional affiliations.
+# All rights reserved.  See the file COPYING in the top-level source
+# directory for licensing information.
+# #############################################################################
+#
+# The build system configuration.
+##
+cvc5_add_unit_test_black(datatype_api_black api)
```
Comment (nafur) [inline test/unit/api/cpp/CMakeLists.txt:15]: Why not move the definition of these macros to this file as well?

File: test/unit/api/python/CMakeLists.txt [hunk 2]
```diff
@@ -17,21 +17,22 @@
 check_python_module("pytest")
 
 # Add Python bindings API tests.
-macro(cvc5_add_python_api_test name filename)
+macro(cvc5_add_python_api_unit_test name filename)
 # We create test target 'python/unit/api/myapitest' and run it with
 # 'ctest -R "python/unit/api/myapitest"'.
-add_test (NAME python/unit/api/${name}
+add_test (NAME ${name}
```
Comment (nafur) [inline test/unit/api/python/CMakeLists.txt:23]: Can you remove the prefix from the Java tests as well for consistency?

Comment (aniemetz) [reply test/unit/api/python/CMakeLists.txt:23]: We decided offline to keep the prefix and use a prefix for all unit tests.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**